prog/postgres/postgres_server_nexus.rb

```diff
@@ -410,14 +435,24 @@ def before_run
 
   label def prepare_for_take_over
     decr_take_over
-    hop_taking_over if postgres_server.resource.representative_server.nil?
+
+    hop_taking_over if postgres_server.read_replica? || postgres_server.resource.representative_server.nil?
```

Comment:
We should increment destroy for representative_server of read replica cluster here as well, no? Do we need this exception for read replicas?

That would also save you same logic from the next label.

Reply:
I think we are still risking some data async issues. Two clients can be connected to different read replicas at the same time and would read different data from each.

Also, I think in both cases, there would be some down time. In the current case, the down time would be much shorter (only connection drop). If we combine the logic, there would be 4-5 seconds of down time, which is acceptable I think.